### PR TITLE
Add MessagePack instances for types without constructors

### PR DESCRIPTION
**Pull request by Martoon-00**

### CODE REVIEW

File: src/Data/MessagePack/Generic.hs [hunk 1]
```diff
@@ -18,6 +18,9 @@ import           GHC.Generics
 import           Data.MessagePack.Class
 import           Data.MessagePack.Object (Object (..))
 
+instance GMessagePack V1 where
+  gToObject = undefined
+  gFromObject = fail "invalid encoding for void type"
```
Comment (Martoon-00) [inline src/Data/MessagePack/Generic.hs]: Probably "can't instantiate void type" would be better here, I just followed `fail` pattern usage all over the project

**Review by Martoon-00 — COMMENTED** [review]
It doesn't build due to a lack of `Data.Void` package, that is strange

File: src/Data/MessagePack/Generic.hs (hunk 1 above)
Comment (iphydf) [reply src/Data/MessagePack/Generic.hs]: I see - there might be a test that checks this. "can't instantiate void type" sounds fine. Are there tests that fail if you do this?

**Comment by iphydf** [comment]
Adding void to the dependencies should fix the build failure.

File: src/Data/MessagePack/Generic.hs (hunk 1 above)
Comment (Martoon-00) [reply src/Data/MessagePack/Generic.hs]: No, there are no tests for this case. As you approve, I'm going to change

**Comment by iphydf** [comment]
Thanks, looks good. Can you rebase on master?

File: src/Data/MessagePack/Generic.hs [hunk 2]
```diff
@@ -18,6 +18,9 @@ import           GHC.Generics
 import           Data.MessagePack.Class
 import           Data.MessagePack.Object (Object (..))
 
+instance GMessagePack V1 where
+  gToObject = undefined
+  gFromObject = fail "can't instantiate void type"
```
Comment (Martoon-00) [inline src/Data/MessagePack/Generic.hs:23]: No tests on error message happened to be for this case

**Review by Martoon-00 — COMMENTED** [review]
What do you mean? Am I not on master?

**Comment by iphydf** [comment]
``` sh
git fetch upstream # assuming toktok/hs-msgpack is your upstream remote
git rebase upstream/master
```

**Comment by Martoon-00** [comment]
Ah, got it. Done

**Review by iphydf — APPROVED** [review]

**Comment by Martoon-00** [comment]
Thanks a lot for your patience )

**Pull request merged**